flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectResultIterator.java

```diff
@@ -40,18 +43,37 @@
 	public CollectResultIterator(
 			CompletableFuture<OperatorID> operatorIdFuture,
 			TypeSerializer<T> serializer,
-			String accumulatorName) {
-		this.fetcher = new CollectResultFetcher<>(operatorIdFuture, serializer, accumulatorName);
+			String accumulatorName,
+			CheckpointConfig checkpointConfig) {
+		if (checkpointConfig.getCheckpointingMode() == CheckpointingMode.EXACTLY_ONCE) {
+			if (checkpointConfig.getCheckpointInterval() >= CheckpointCoordinatorConfiguration.MINIMAL_CHECKPOINT_TIME) {
+				this.fetcher = new CollectResultFetcher<>(
+					new CheckpointedCollectResultBuffer<>(serializer),
+					operatorIdFuture,
+					accumulatorName);
+			} else {
+				this.fetcher = new CollectResultFetcher<>(
+					new UncheckpointedCollectResultBuffer<>(serializer, false),
+					operatorIdFuture,
+					accumulatorName);
+			}
```

Comment:
If `checkpointConfig.getCheckpointInterval() >= CheckpointCoordinatorConfiguration.MINIMAL_CHECKPOINT_TIME` we are sure that the user explicitly enables a checkpoint. Otherwise we have to sync with the default value of checkpoint interval in `CheckpointCoordinatorConfiguration`.

Reply:
OK... It seems that `checkpointConfig.isCheckpointingEnabled()` is a better solution.